modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -189,21 +203,30 @@ public MetadataMediator(T componentModel) {
 
     MetadataResult<OutputMetadataDescriptor> output = outputDelegate.getOutputMetadataDescriptor(context, keyValue);
     MetadataResult<InputMetadataDescriptor> input = inputDelegate.getInputMetadataDescriptors(context, keyValue);
-
-    if (output.isSuccess() && input.isSuccess()) {
+    Optional<MetadataResult<InputMetadataDescriptor>> successCallbackInput = successCallbackInputDelegate
+        .map(successCallbackInputDelegate -> successCallbackInputDelegate.getInputMetadataDescriptors(context, keyValue));
+    Optional<MetadataResult<InputMetadataDescriptor>> errorCallbackInput = errorCallbackInputDelegate
+        .map(errorCallbackInputDelegate -> errorCallbackInputDelegate.getInputMetadataDescriptors(context, keyValue));
+    if (output.isSuccess() && input.isSuccess() && (!successCallbackInput.isPresent() || successCallbackInput.get().isSuccess())
+        &&
+        (errorCallbackInput.isPresent() || errorCallbackInput.get().isSuccess())) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "errorCallbackInput.isPresent()" before accessing the value. ([squid:S3655](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7668&issues=AWjdfEMLotIXnmsUGfbW&open=AWjdfEMLotIXnmsUGfbW)

[ISSUE_KEY]: AWjdfEMLotIXnmsUGfbW

Reply:
@ndinu what about this?